client/src/app/hooks/useIsArchetype.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+
+const useIsArchetype = () => {
+  const location = useLocation();
+  const [isArchetype, setIsArchetype] = useState(
+    location.pathname.includes("/archetypes/")
+  );
+
+  useEffect(() => {
+    setIsArchetype(location.pathname.includes("/archetypes/"));
+  }, [location.pathname]);
```

Comment:
You don't need the state or the effect here because all they are doing is duplicating the new value from `useLocation()` when it causes a re-render. If that `useEffect` is being triggered, it means we already have an updated `location.pathname` value in scope (because `useLocation` watches for it to change and re-renders), and we can just return that directly instead of putting it in state. So I think this whole hook can be rewritten as:

```ts
const useIsArchetype = () => useLocation().pathname.includes("/archetypes/");
```

Reply:
Updated - thanks!